src/Microsoft.AspNetCore.WebUtilities/MultipartReaderStream.cs

```diff
@@ -40,15 +38,8 @@ public MultipartReaderStream(BufferedReadStream stream, string boundary, bool ex
 
             _innerStream = stream;
             _innerOffset = _innerStream.CanSeek ? _innerStream.Position : 0;
-            if (expectLeadingCrlf)
-            {
-                _boundaryBytes = Encoding.UTF8.GetBytes("\r\n--" + boundary);
-            }
-            else
-            {
-                _boundaryBytes = Encoding.UTF8.GetBytes("--" + boundary);
-            }
-            _finalBoundaryLength = _boundaryBytes.Length + 2; // Include the final '--' terminator.
+            _boundary = boundary;
+            _boundary.ExpectLeadingCrlf = expectLeadingCrlf;
```

Comment:
It's weird to take a constructor parameter that only modifies another parameter, especially one that already takes the same parameter.